packages/cxl-ui/src/components/cxl-playbook-accordion.js

```diff
@@ -0,0 +1,163 @@
+  }
+
+  static _getAccordionPanelStyles() {
+    return css`
+      [part='summary'] {
+        box-shadow: 0 4px 12px rgba(36, 38, 40, 0.12);
+        padding: 0;
+      }
+      :host(.emptyContent) [part='toggle'] {
+        display: none;
+      }
+      [part='toggle'] {
+        padding: 12px;
+      }
+    `;
+  }
+
+  static _getVaadinCheckboxStyles() {
+    return css`
+      :host([theme~='custom']) [part='checkbox'] {
+        border: 2px solid !important;
+        width: 18px;
+        height: 18px;
+      }
+      :host([theme~='custom']) [part='label'] {
+        display: none;
+      }
+    `;
+  }
+
+  /**
+   * Global styles.
+   */
+  ready() {
+    super.ready();
+
+    // Define and register a style sheet for the <vaadin-text-field> component
+    registerGlobalStyles(cxlAdeftAccordionGlobalStyles, {
+      moduleId: 'cxl-playbook-accordion-global',
+    });
+
+    this._updateCheckboxesStates();
+  }
+
+  /**
+   *
+   * @param {HTMLCollection} items
+   * @private
+   */
+  _saveAccordionState(items) {
+    const storageId = this.getAttribute('id');
+    // Avoid null key.
+    if (storageId) {
+      const stateItems = [];
+
+      items.forEach((value, key) => {
+        stateItems[key] = items[key].opened;
+      });
+
+      localStorage.setItem(storageId, JSON.stringify(stateItems));
+
+      this._updateCSSAndPanelStateToCheckboxesStates();
+
+      this._dispatchCustomEvent(stateItems);
+
+      this._saveCheckboxesState();
+    }
+  }
+
+  _saveCheckboxesState() {
+    const stateCheckboxes = [];
+    const checkboxes = this.querySelectorAll('vaadin-checkbox');
+
+    checkboxes.forEach((value, key) => {
+      const checkbox = checkboxes[key];
+
+      stateCheckboxes[key] =
+        checkbox.hasAttribute('aria-checked') && checkbox.getAttribute('aria-checked') === 'true';
+    });
+
+    const checkboxesStorageId = `${this.getAttribute('id')}_checkboxes`;
+    localStorage.setItem(checkboxesStorageId, JSON.stringify(stateCheckboxes));
+  }
+
+  // Restore accordion panel state.
+  _updateItems(items, opened) {
+    super._updateItems(items, opened);
+  }
+
+  _updateCheckboxesStates() {
+    const checkboxesStorageId = `${this.getAttribute('id')}_checkboxes`;
+    const stateCheckboxes = JSON.parse(localStorage.getItem(checkboxesStorageId));
+
+    if (stateCheckboxes === null) {
+      return;
+    }
+
+    const checkboxes = this.querySelectorAll('vaadin-checkbox');
+
+    checkboxes.forEach((item, key) => {
+      const checkbox = checkboxes[key];
+      const isChecked = !!stateCheckboxes[key]; // autocast from null or undefined to boolean
+      checkbox.setAttribute('aria-checked', isChecked ? 'true' : 'false');
+      checkbox.checked = isChecked;
+    });
+
+    this._updateCSSAndPanelStateToCheckboxesStates();
+  }
+
+  _updateCSSAndPanelStateToCheckboxesStates() {
+    const checkboxes = this.querySelectorAll('vaadin-checkbox');
```

Comment:
DRY is a primary principle.

Every time you see yourself repeating things, find a way to refactor and centralize.

Reply:
code refactored